examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Random.hpp>
+
+template <class MemorySpace>
+struct Neighbors
+{
+  Kokkos::View<ArborX::Point *, MemorySpace> particles_;
```

Comment:
Maybe template on `Particles`, so we could experiment with layouts. For example, if an application has separate `x`, `y`, `z`?